pkg/util/resource/resource.go

```diff
@@ -103,3 +103,14 @@ func QuantityToFloat(q *resource.Quantity) float64 {
 	}
 	return float64(q.MilliValue()) / 1000
 }
+
+// MulByFloat multiplies every element in q by f.
+// Leverages k8s.io/apimachinery/pkg/api/resource package to provide precision to 3 decimal places
+func MulByFloat(q corev1.ResourceList, f float64) corev1.ResourceList {
+	ret := q.DeepCopy()
+	for k, v := range ret {
+		scaledV := float64(v.MilliValue()) * f
+		ret[k] = *resource.NewMilliQuantity(int64(scaledV), resource.DecimalSI)
+	}
+	return ret
+}
```

Comment:
Could we detect overflow and underflow?
The k/k resource `Mul` accommodates only int values since we can avoid unexpected overflow and underflow.
However, this does not have any guards.

Reply:
The `f` multiplier is always in range [0,1] so the overflow won't occur. When it comes to underflow, go `float64` provides ~15 decimal digits precision, and this is definitely sufficient for our use case. It the underflow happens and the result will be rounded to 0, I believe it's WAI. Given that we have to convert it to `int64` eventually, we will still loose some precision, but I think we're good with it, as the feature itself doesn't need to be really precise, but rather present some notion of LQ's usage

Reply:
> The f multiplier is always in range [0,1] so the overflow won't occur.

Yes, you're right. The key problem is this is exposed function and generic one. Actually, this function name does not imply that the input range should be [0, 1]. So, we can consider 1. adding validations for  input `f` or 2. adding just comments that this expects [0, 1], otherwise, it will occur unexpected overflow.

I think 2 would be enough since 1 will decrease usability.

Reply:
Agree, will add the comment